src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -61,6 +74,10 @@ public JsonAdaptedPerson(Person source) {
                 .map(JsonAdaptedTag::new)
                 .collect(Collectors.toList()));
         remark = source.getRemark().value;
+        dateOfCreation = source.getDateOfCreation().toString();
+        historyEntries.addAll(source.getHistory().getHistoryEntries().entrySet().stream()
```

Comment:
I don't fully know how this works yet, someone else can verify

Reply:
this is just to convert a treemap to lists and add everything to historyEntries